hail/src/main/scala/is/hail/utils/SpillingCollectIterator.scala

```diff
@@ -0,0 +1,97 @@
+package is.hail.utils
+
+import is.hail.HailContext
+import java.io.{ ObjectInputStream, ObjectOutputStream }
+import org.apache.spark.rdd.RDD
+import scala.reflect.ClassTag
+import scala.reflect.classTag
+
+object SpillingCollectIterator {
+  def apply[T: ClassTag](rdd: RDD[T], sizeLimit: Int): SpillingCollectIterator[T] = {
+    val x = new SpillingCollectIterator(rdd, sizeLimit)
+    val ctc = classTag[T]
+    HailContext.get.sc.runJob(
+      rdd,
+      (_, it: Iterator[T]) => it.toArray(ctc),
+      0 until rdd.partitions.length,
+      x.append _)
+    x
+  }
+}
+
+class SpillingCollectIterator[T: ClassTag] private (rdd: RDD[T], sizeLimit: Int) extends Iterator[T] {
```

Comment:
This needs unit tests -- can use property-based testing to generate random arrays, partitioning, size limits, then compare `array.iterator` with `SpillingCollectIterator(sc.parallelize(array, nPartitions), sizeLimit)`